rmf_site_format/src/scenario.rs

```diff
@@ -19,13 +19,78 @@ use crate::*;
 #[cfg(feature = "bevy")]
 use bevy::prelude::{Bundle, Component, Reflect, ReflectComponent};
 use serde::{Deserialize, Serialize};
-use std::collections::HashMap;
+use std::collections::{BTreeMap, HashMap};
 
 #[derive(Serialize, Deserialize, Debug, Clone, Copy, Default, PartialEq, Eq)]
 #[cfg_attr(feature = "bevy", derive(Component, Reflect))]
 #[cfg_attr(feature = "bevy", reflect(Component))]
 pub struct InstanceMarker;
 
+#[derive(Serialize, Deserialize, Debug, Clone, PartialEq)]
+#[cfg_attr(feature = "bevy", derive(Component))]
+pub enum InstanceModifier {
+    Added(AddedInstance),
+    Inherited(InheritedInstance),
+    Hidden,
+}
+
+impl InstanceModifier {
+    pub fn new_added(pose: Pose) -> Self {
+        Self::Added(AddedInstance { pose: pose })
+    }
+
+    pub fn new_inherited(pose: Option<Pose>) -> Self {
```

Comment:
Let's do

```rust
pub fn inherited() -> Self {
```

I don't think we're likely to see scenarios where we're creating a new `Inherited` from scratch and immediately have a modified pose for it.

Reply:
We'd need to create new `Inherited` with modified poses when toggling instance modifiers from `Hidden` to `Inherited` in cases where the model instance has been moved and toggled off.

Reply:
I ended up following your suggestion to take out the arguments, and in cases where we need to instantiate with values I created `InheritedInstance { .. }` directly instead to accommodate future modifications.